src/components/authors/manageAuthorPage.js

```diff
@@ -1,46 +1,37 @@
 "use strict";
 
 var React = require('react');
-var Router = require('react-router');
+var createReactClass = require('create-react-class');
+var ReactRouter = require('react-router-dom');
 var AuthorForm = require('./authorForm');
 var AuthorActions = require('../../actions/authorActions');
 var AuthorStore = require('../../stores/authorStore');
 var toastr = require('toastr');
 
-var ManageAuthorPage = React.createClass({
-	mixins: [
-		Router.Navigation
-	],
-
-	statics: {
-		willTransitionFrom: function(transition, component) {
-			if (component.state.dirty && !confirm('Leave without saving?')) {
-				transition.abort();
-			}
-		}
-	},
+var Redirect = ReactRouter.Redirect;
+var Prompt = ReactRouter.Prompt;
 
+var ManageAuthorPage = createReactClass({
 	getInitialState: function() {
 		return {
 			author: { id: '', firstName: '', lastName: '' },
 			errors: {},
-			dirty: false
+			dirty: false,
+			redirect: false
 		};
 	},
 
-	componentWillMount: function() {
-		var authorId = this.props.params.id; //from the path '/author:id'
+	componentDidMount: function() {
+		var authorId = this.props.match.params.id; //from the path '/author:id'

```

Comment:
It's best practice to fire AJAX calls in `componentDidMount`. Also, referencing `this.props.match` since that's where params are exposed by React Router 4.